packages/react-reconciler/src/ReactFiberHooks.new.js

```diff
@@ -723,7 +761,14 @@ function createFunctionComponentUpdateQueue(): FunctionComponentUpdateQueue {
 }
 
 function useMemoCache(size: number): Array<any> {
-  throw new Error('Not implemented.');
+  let memoCache = currentlyRenderingFiber.memoCache;
+  if (memoCache === null) {
+    memoCache = currentlyRenderingFiber.memoCache = {
+      current: new Array(size),
+      previous: null,
+    };
+  }
+  return memoCache.current;
```

Comment:
Like `useContext` this hook isn't actually using the internal hooks data structure. This seems functional but it does add a Fiber property that is only useful for one Fiber type (Function components) and perhaps using the hooks infra would be more appropriate (I'm not actually sure about this feedback, just food for thought)

If you modelled this more like a traditional hook you would have a mount and update implementation that get a hook using `mountWorkInProgressHook` and `updateWorkInProgressHook` respectively. you can then store the cache as the hooks `.memoizedState` and clone it on update.

If you go this route the Fiber doesn't need a new property and you don't need to null check the cache because it will always be there when used.

the simplest analagous hook methods might be here:
https://github.com/facebook/react/blob/main/packages/react-reconciler/src/ReactFiberHooks.old.js#L2129-L2141

Reply:
Also this feedback would eliminate the need for the feedback in the other comment since you would drop the memoCache fiber property altogether

Reply:
The way I read this, this also just supports a single `useMemoCache` call per component? Note that we might have multiple caches per component when custom hooks have their own memo cache.